src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,224 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    private static class Node<T> {
+        private T item;
+        private Node<T> next;
+        private Node<T> previous;
+
+        private Node(T element) {
+            this.item = element;
+        }
+
+        private Node(Node<T> previous, T element, Node<T> next) {
+            this.item = element;
+            this.next = next;
+            this.previous = previous;
+        }
+    }
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            Node<T> newNode = new Node<>(null, value, null);
+            head = tail = newNode;
+        } else {
+            Node<T> newNode = new Node<>(tail, value, null);
+            tail.next = newNode;
+            tail = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (size == 0 && index == size) {
+            add(value);
+            return;
+        }
+        if (size != 0 && index == size) {
+            add(value);
+            return;
+        }
+        if (index > 0 && index < size) {
+            Node<T> currentNode = findNode(index);
+            Node<T> addNode = new Node<>(null, value, currentNode);
+            addNode.previous = currentNode.previous;
+            currentNode.previous.next = addNode;
+            currentNode.previous = addNode;
+            size++;
+            return;
+        }
+        if (index == 0 && index < size) {
+            Node<T> currentNode = findNode(index);
+            Node<T> addNode = new Node<>(null, value, currentNode);
+            addNode.previous = currentNode.previous;
+            currentNode.previous = addNode;
+            head = addNode;
+            size++;
+            return;
+        }
+
+        if (index <= 0 || index > size) {
+            throw new IndexOutOfBoundsException("Wrong Index.");
```

Comment:
It would be much better to create checkIndex(int index) method with logic which you used here and then just call it